lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -1178,43 +1333,39 @@ def decode_cloudfunctions_function_name(function_name)
         .gsub('u.u', '_').gsub('d.d', '$').gsub('a.a', '@').gsub('p.p', '.')
     end
 
-    # Requires that record has a 'kubernetes' field.
-    def extract_container_metadata(record)
-      resource_labels = {}
-      common_labels = {}
-      %w(namespace_id pod_id container_name).each do |field|
-        resource_labels.merge!(
-          fields_to_labels(record['kubernetes'], field => field))
-      end
-      %w(namespace_name pod_name).each do |field|
-        common_labels.merge!(
-          fields_to_labels(
-            record['kubernetes'],
-            field => "#{CONTAINER_CONSTANTS[:service]}/#{field}"))
-      end
-      # Prepend label/ to all user-defined labels' keys.
-      if record['kubernetes'].key?('labels')
-        record['kubernetes']['labels'].each do |key, value|
-          common_labels["label/#{key}"] = value
-        end
+    def format(tag, time, record)
+      [tag, time, record].to_msgpack
+    end
+
+    # Given a tag, returns the corresponding valid tag if possible, or nil if
+    # the tag should be rejected. If 'require_valid_tags' is false, non-string
+    # tags are converted to strings, and invalid characters are sanitized;
+    # otherwise such tags are rejected.
+    def sanitize_tag(tag)
+      if @require_valid_tags &&
+         (!tag.is_a?(String) || tag == '' || convert_to_utf8(tag) != tag)
+        return nil
       end
-      # We've explicitly consumed all the fields we care about -- don't litter
-      # the log entries with the remaining fields that the kubernetes metadata
-      # filter plugin includes (or an empty 'kubernetes' field).
-      record.delete('kubernetes')
-      record.delete('docker')
-      [resource_labels, common_labels]
+      tag = convert_to_utf8(tag.to_s)
+      tag = '_' if tag == ''
+      tag
     end
 
     # For every original_label => new_label pair in the label_map, delete the
-    # original_label from the record if it exists, and extract the value to form
-    # a map with the new_label as the key.
-    def fields_to_labels(record, label_map)
-      return {} if label_map.nil? || !label_map.is_a?(Hash)
+    # original_label from the hash map if it exists, and extract the value to
+    # form a map with the new_label as the key.
+    def delete_and_extract_labels(hash, label_map, should_delete = true)
```

Comment:
Ah, I added this because in order to apply `delete_and_extract_labels` to `matched_regex_group`. Right now we are setting matched_regex_group['pod_name'] to both resource label `pod_id` and common label `container.googleapis.com/pod_name`. So we can not destroy the hash on the first run.